### ISSUE TITLE
Struck at wait(self, timeout) in threading.py during model training on Google Colab

### ISSUE THREAD

**Issue by Shreyas-Gupta-21** [matterport/Mask_RCNN#2696]
During training the Mask RCNN on custom dataset, I am getting following error - 

/content/drive/MyDrive/Colab/Mask_RCNN/mrcnn/model.py in train(self, train_dataset, val_dataset, learning_rate, epochs, layers, augmentation, custom_callbacks, no_augmentation_sources)
   2372             max_queue_size=100,
   2373             workers=workers,
-> 2374             use_multiprocessing=workers > 1,
   2375         )
   2376         self.epoch = max(self.epoch, epochs)

/usr/local/lib/python3.7/dist-packages/keras/engine/training_v1.py in fit(self, x, y, batch_size, epochs, verbose, callbacks, validation_split, validation_data, shuffle, class_weight, sample_weight, initial_epoch, steps_per_epoch, validation_steps, validation_freq, max_queue_size, workers, use_multiprocessing, **kwargs)
    794         max_queue_size=max_queue_size,
    795         workers=workers,
--> 796         use_multiprocessing=use_multiprocessing)
    797 
    798   def evaluate(self,

/usr/local/lib/python3.7/dist-packages/keras/engine/training_generator_v1.py in fit(self, model, x, y, batch_size, epochs, verbose, callbacks, validation_split, validation_data, shuffle, class_weight, sample_weight, initial_epoch, steps_per_epoch, validation_steps, validation_freq, max_queue_size, workers, use_multiprocessing)
    584         shuffle=shuffle,
    585         initial_epoch=initial_epoch,
--> 586         steps_name='steps_per_epoch')
    587 
    588   def evaluate(self,

/usr/local/lib/python3.7/dist-packages/keras/engine/training_generator_v1.py in model_iteration(model, data, steps_per_epoch, epochs, verbose, callbacks, validation_data, validation_steps, validation_freq, class_weight, max_queue_size, workers, use_multiprocessing, shuffle, initial_epoch, mode, batch_size, steps_name, **kwargs)
    210     step = 0
    211     while step < target_steps:
--> 212       batch_data = _get_next_batch(generator)
    213       if batch_data is None:
    214         if is_dataset:

/usr/local/lib/python3.7/dist-packages/keras/engine/training_generator_v1.py in _get_next_batch(generator)
    344   """Retrieves the next batch of input data."""
    345   try:
--> 346     generator_output = next(generator)
    347   except (StopIteration, tf.errors.OutOfRangeError):
    348     return None

/usr/local/lib/python3.7/dist-packages/keras/utils/data_utils.py in get(self)
    771     while self.is_running():
    772       try:
--> 773         inputs = self.queue.get(block=True, timeout=5).get()
    774         if self.is_running():
    775           self.queue.task_done()

/usr/lib/python3.7/queue.py in get(self, block, timeout)
    177                     if remaining <= 0.0:
    178                         raise Empty
--> 179                     self.not_empty.wait(remaining)
    180             item = self._get()
    181             self.not_full.notify()

/usr/lib/python3.7/threading.py in wait(self, timeout)
    298             else:
    299                 if timeout > 0:
--> 300                     gotit = waiter.acquire(True, timeout)
    301                 else:
    302                     gotit = waiter.acquire(False)

**Comment by josemarcosrf** [comment]
@Shreyas-Gupta-21 I am facing the same issue.

What versions of tensorflow / keras are you using?

**Comment by leon-kwy** [comment]
> 

Do you fix this probelm? I'm facing the same issue as well.

**Comment by josemarcosrf** [comment]
As a work-around I changed the [line 2362](https://github.com/matterport/Mask_RCNN/blob/3deaec5d902d16e1daf56b62d5971d428dc920bc/mrcnn/model.py#L2362) to `workers = 1` instead.

Passing `workers=1` directly in the keras fit_generator call would be equivalent. i.e.: [here](https://github.com/matterport/Mask_RCNN/blob/3deaec5d902d16e1daf56b62d5971d428dc920bc/mrcnn/model.py#L2373)

In this way the multi-processing part in keras doesn't seem to get stuck....
